Tools/AP_Periph/can.cpp

```diff
@@ -1950,6 +1958,16 @@ void AP_Periph_FW::can_update()
 
         }
     }
+
+    // if there is a reboot scheduled, do it 1 second after request to allow the acknowledgement to be sent
```

Comment:
1 second seems a little arbitrary.  And perhaps long.

It would be nice to have some sort of trigger (ie. triggered when we know the packet's been successfully sent), with the timeout as a secondary trigger. 

Probably too much work, really.